<a href="https://colab.research.google.com/github/RogerNauer/DSToolbox/blob/master/RogelioMateos_Activity1_CollabJupyter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Activity #1**

Rogelio Mateos A00171369

First activity to select a dataset for data mining exercises.

In [1]:
import pandas as pd

In [2]:
path='/content/drive/MyDrive/Colab Notebooks/Telco_customer_churn.csv'

In [3]:
df=pd.read_csv(path)
df.head()

CustomerID  Count        Country       State         City  Zip Code  \
0  3668-QPYBK      1  United States  California  Los Angeles     90003   
1  9237-HQITU      1  United States  California  Los Angeles     90005   
2  9305-CDSKC      1  United States  California  Los Angeles     90006   
3  7892-POOKP      1  United States  California  Los Angeles     90010   
4  0280-XJGEX      1  United States  California  Los Angeles     90015   

                 Lat Long   Latitude   Longitude  Gender  ...        Contract  \
0  33.964131, -118.272783  33.964131 -118.272783    Male  ...  Month-to-month   
1   34.059281, -118.30742  34.059281 -118.307420  Female  ...  Month-to-month   
2  34.048013, -118.293953  34.048013 -118.293953  Female  ...  Month-to-month   
3  34.062125, -118.315709  34.062125 -118.315709  Female  ...  Month-to-month   
4  34.039224, -118.266293  34.039224 -118.266293    Male  ...  Month-to-month   

  Paperless Billing             Payment Method  Monthly Charges Total Charges  \
0               Yes               Mailed check            53.85        108.15   
1               Yes           Electronic check            70.70        151.65   
2               Yes           Electronic check            99.65         820.5   
3               Yes           Electronic check           104.80       3046.05   
4               Yes  Bank transfer (automatic)           103.70        5036.3   

  Churn Label Churn Value Churn Score  CLTV                   Churn Reason  
0         Yes           1          86  3239   Competitor made better offer  
1         Yes           1          67  2701                          Moved  
2         Yes           1          86  5372                          Moved  
3         Yes           1          84  5003                          Moved  
4         Yes           1          89  5340  Competitor had better devices  

[5 rows x 33 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 33 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   CustomerID         7043 non-null   object 
 1   Count              7043 non-null   int64  
 2   Country            7043 non-null   object 
 3   State              7043 non-null   object 
 4   City               7043 non-null   object 
 5   Zip Code           7043 non-null   int64  
 6   Lat Long           7043 non-null   object 
 7   Latitude           7043 non-null   float64
 8   Longitude          7043 non-null   float64
 9   Gender             7043 non-null   object 
 10  Senior Citizen     7043 non-null   object 
 11  Partner            7043 non-null   object 
 12  Dependents         7043 non-null   object 
 13  Tenure Months      7043 non-null   int64  
 14  Phone Service      7043 non-null   object 
 15  Multiple Lines     7043 non-null   object 
 16  Internet Service   7043 

**Name: Telco Customer Churn Dataset**

Location: This dataset is available in Kaggle at https://www.kaggle.com/datasets/yeanzc/telco-customer-churn-ibm-dataset

Organization that generated: IBM generated this dataset for analytical purposes based on a fictional Telco company and a list of customers labeled with churn status.

**Content of the dataset**

It contains 7,043 records with a total of 33 columns.

The most relevant columns are:

CustomerID & the Churn Label (Yes/No), which allow us to differentiate two types of customers, those who remain in the compañy and those who left.

Also there are some other columns like:

Gender (male/female), Dependents (Yes/No) to know if the customer lives with other people where he is responsible, Tenure Months to indicate how many months the client was with the company services.

And there are a few columns to indicate the customer services contracted during his time like: Phone Services, Multiple Lines, Internet Service, Online Security, etc.

There are also some columns about the type of contract of the customer: Contract (monthly, annual, etc), Payment Method (deposit, credit card, mailed check, etc), Monthly Charge and Total Charges.

There are some additional columns for Geolocation which are very important like: State, City, Country, Zip Code, Longitude & Latitude.

**Research questions**

Who are our most valuable customers?
Can we segment or customer value by address or location?
Is there any services issues correlation with customer churn?
Which of our customers are more likely to churn pretty soon?
Which of our customers might be open to an offer of increase of additional services without causing churn in the future?
What are the main reasons customers are leaving our company?
Can it be designed a model to predict our customers churn and what would be its accuracy?
Which level of incentives can be offered for retention but balancing the expense in such offers?

**Stakeholders interested**

Theses insights would be very valuable for the Telco companies for having a prediction model to design, plan and drive a retention system. Also by designing this model it could be translated to other type of companies following the same basic recipe but tuning differente attributes of data, specially for companies of services.

Identify who are our most valuable customers and predict which ones are about to leave would support almost any retail or services company for three main reasons:

Increase retention and reduce expense for adquiring new customers.

Identify root issues or locations where there might be some service quality problems that might be impacting customers satisfaction.

Review current company plans and charges in case of any correlation with the plans offering vs other companies.